paddle/fluid/framework/threadpool.h

```diff
@@ -84,7 +83,13 @@ class ThreadPool {
       return nullptr;
     });
     std::future<std::unique_ptr<platform::EnforceNotMet>> f = task.get_future();
-    tasks_.push(std::move(task));
+    {
+      std::unique_lock<std::mutex> lock(mutex_);
+      if (!running_) {
```

Comment:
Is this `running_` check needed?

Reply:
I think we need it, but it can work if not add.